doc/source/gapfill.md

```diff
@@ -0,0 +1,272 @@
+glaciers_2010 = gu.Vector(xdem.examples.get_path("longyearbyen_glacier_outlines_2010"))
+
+# Make a dictionary of glacier outlines where the key represents the associated date.
+outlines = {
+    datetime(1990, 8, 1): glaciers_1990,
+    datetime(2009, 8, 1): glaciers_2010,
+}
+
+# Cropping DEMs to a smaller extent to visualize the gap-filling better
+bounds = (dem_2009.bounds.left, dem_2009.bounds.bottom,
+          dem_2009.bounds.left + 200 * dem_2009.res[0], dem_2009.bounds.bottom + 150 * dem_2009.res[1])
+dem_2009 = dem_2009.crop(bounds)
+dem_1990 = dem_1990.crop(bounds)
+```
+
+We create a difference of DEMs object {class}`xdem.ddem.dDEM` to experiment on:
+
+```{code-cell} ipython3
+ddem = xdem.dDEM(raster=dem_2009 - dem_1990, start_time=dem_1990.datetime, end_time=dem_2009.datetime)
+
+# The example DEMs are void-free, so let's make some random voids.
+# Introduce 50000 nans randomly throughout the dDEM.
+mask = np.zeros_like(ddem.data, dtype=bool)
+mask.ravel()[(np.random.choice(ddem.data.size, int(ddem.data.size/5), replace=False))] = True
+ddem.set_mask(mask)
+```
+
+## Linear spatial interpolation
+
+Linear spatial interpolation (also often called bilinear interpolation) of dDEMs is arguably the simplest approach: voids are filled by an average of the surrounding pixels values, weighted by their distance to the void pixel.
+
+```{code-cell} ipython3
+ddem_linear = ddem.interpolate(method="linear")
+```
+
+```{code-cell} ipython3
+:tags: [hide-input]
+:mystnb:
+:  code_prompt_show: "Show plotting code"
+:  code_prompt_hide: "Hide plotting code"
+
+ddem_linear = ddem.copy(new_array=ddem_linear)
+
+# Plot before and after
+f, ax = plt.subplots(1, 2)
+ax[0].set_title("Before linear\ngap-filling")
+ddem.plot(cmap='RdYlBu', vmin=-10, vmax=10, ax=ax[0])
+ax[1].set_title("After linear\ngap-filling")
+ddem_linear.plot(cmap='RdYlBu', vmin=-10, vmax=10, ax=ax[1], cbar_title="Elevation differences (m)")
+_ = ax[1].set_yticklabels([])
+```
+
+## Local hypsometric interpolation
+
+This approach assumes that there is a relationship between the elevation and the elevation change in the dDEM, which is often the case for glaciers.
+Elevation change gradients in late 1900s and 2000s on glaciers often have the signature of large melt in the lower parts, while the upper parts might be less negative, or even positive.
+This relationship is strongly correlated for a specific glacier, and weakly correlated on regional scales.
+With the local (glacier specific) hypsometric approach, elevation change gradients are estimated for each glacier separately.
+This is simply a linear or polynomial model estimated with the dDEM and a reference DEM.
+Then, voids are interpolated by replacing them with what "should be there" at that elevation, according to the model.
+
+```{code-cell} ipython3
+ddem_localhyps = ddem.interpolate(method="local_hypsometric", reference_elevation=dem_2009, mask=glaciers_1990)
```

Comment:
The API for the function `interpolate` is not available, so it is difficult to get detailed info on the arguments. I was actually wondering about the type of argument `mask` as it looks like you are passing a Vector and not a mask.
I know this module needs more work, but good to raise an issue if this is not already mentioned.
I was also thinking that in the same way you generalized the bias correction to take any predictor variable, we could generalize the hypsometric method to work with any 1, 2 or Nd variables.

Reply:
As it's glacier-focused, I think we need to discuss removing it altogether in its current state, including potentially the `dDEM` class. I've already added warning everywhere (API, pages and gallery examples) that these might be removed/re-factored.
Opening #636